src/py/tests/test_page_generator.py

```
@@ -155,37 +189,256 @@ async def test_page_generator():
             "or if you really need to, run it separately and then skip it "
             "in the main group.",
         )
+
+    # Test no imports (plotly not available)
     no_imports = PageGenerator().generate_index()
-    assert no_imports_result_re.findall(no_imports), (
-        f"{len(no_imports_result_raw)}: {no_imports_result_raw}"
-        "\n"
-        f"{len(no_imports)}: {no_imports}"
-    )
+    scripts, encodings = get_scripts_from_html(no_imports)
+
+    # Should have mathjax, plotly default, and kaleido_scopes
+    assert len(scripts) == 3  # noqa: PLR2004
+    assert scripts[0] == DEFAULT_MATHJAX
+    assert scripts[1] == DEFAULT_PLOTLY
+    assert scripts[2].endswith("kaleido_scopes.js")
+
     sys.path = old_path
 
-    # this imports plotly so above test must have already been done
+
+async def test_defaults_with_plotly_available():
+    """Test defaults when plotly package is available."""
     all_defaults = PageGenerator().generate_index()
-    assert all_defaults_re.findall(all_defaults)
+    scripts, encodings = get_scripts_from_html(all_defaults)
 
-    with_plot = PageGenerator(plotly="https://with_plot").generate_index()
-    assert with_plot_result_re.findall(with_plot)
+    # Should have mathjax, plotly package data, and kaleido_scopes
+    assert len(scripts) == 3  # noqa: PLR2004
+    assert scripts[0] == DEFAULT_MATHJAX
+    assert scripts[1].endswith("package_data/plotly.min.js")
+    assert scripts[2].endswith("kaleido_scopes.js")
+
+
+async def test_force_cdn():
+    """Test force_cdn=True forces use of CDN plotly even when plotly is available."""
+    # Verify plotly is available first
+    if not find_spec("plotly"):
+        pytest.skip("Plotly not available - cannot test force_cdn override")
```

Shouldn't this test _only_ run when `plotly` is available for import?

---

Yes, here it says if it can't find plotly, it skips the test. Or "only run when plotly is available"

---

Yeah what I mean is that I don't like the usage of `if (condition): pytest.skip()` because it could lead to essentially silently skipping this test entirely if Plotly is never available when the test is run. 

Can't this `if` check be removed entirely?

---

Okay, but I did it in a later PR